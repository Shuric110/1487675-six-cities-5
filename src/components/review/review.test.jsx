import React from "react";
import renderer from "react-test-renderer";

import Review from "./review";

const reviews = [
  {
    id: 1,
    authorAvatar: `/img/avatar-angelina.jpg`,
    authorName: `Angelina`,
    rating: 4,
    date: new Date(`2020-10-17T03:24:00`),
    text: `We enjoyed our stay in the hotel. It was a very cozy atmosphere.`,
  },
  {
    id: 2,
    authorAvatar: `/img/avatar-max.jpg`,
    authorName: `Max`,
    rating: 1,
    date: new Date(`2019-11-19T02:54:10`),
    text: `There were roaches in the closet, several crawling the wall, several came from behind the mirror, and to top it off, found some dead ones in the refrigerator.`,
  },
];

describe(`Render Review`, () => {
  it(`Review for review variant 1 renders correctly`, () => {
    const tree = renderer
      .create(
          <Review
            review={reviews[0]}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Review for review variant 2 renders correctly`, () => {
    const tree = renderer
      .create(
          <Review
            review={reviews[1]}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
