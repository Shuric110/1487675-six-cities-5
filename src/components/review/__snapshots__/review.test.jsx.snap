// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Review Review for review variant 1 renders correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="/img/avatar-angelina.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Angelina
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      We enjoyed our stay in the hotel. It was a very cozy atmosphere.
    </p>
    <time
      className="reviews__time"
      dateTime="2020-10-17"
    >
      October 2020
    </time>
  </div>
</li>
`;

exports[`Render Review Review for review variant 2 renders correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="/img/avatar-max.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Max
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "20%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      There were roaches in the closet, several crawling the wall, several came from behind the mirror, and to top it off, found some dead ones in the refrigerator.
    </p>
    <time
      className="reviews__time"
      dateTime="2019-11-19"
    >
      November 2019
    </time>
  </div>
</li>
`;
